java/pom.xml

```diff
@@ -61,9 +61,9 @@
     <version>1.1.0-SNAPSHOT</version>
 
     <properties>
-        <java.version>17</java.version>
+        <java.version>21</java.version>
```

Comment:
no problem with maven version ?

Reply:
We will revert this to 17 in this pr to keep java 17 source compatibility, even using jdk 21